pkg/controller/installation/validation.go

```diff
@@ -130,22 +130,37 @@ func validateCustomResource(instance *operatorv1.Installation) error {
 			}
 
 			if instance.Spec.CNI.Type == operatorv1.PluginCalico {
-				// Verify the specified encapsulation type is valid.
-				switch v4pool.Encapsulation {
-				case operatorv1.EncapsulationIPIP, operatorv1.EncapsulationIPIPCrossSubnet:
-					// IPIP currently requires BGP to be running in order to program routes.
-					if instance.Spec.CalicoNetwork.BGP == nil || *instance.Spec.CalicoNetwork.BGP == operatorv1.BGPDisabled {
-						return fmt.Errorf("IPIP encapsulation requires that BGP is enabled")
+				if instance.Spec.CNI.IPAM.Type == operatorv1.IPAMPluginCalico {
+					// Verify the specified encapsulation type is valid.
+					switch v4pool.Encapsulation {
+					case operatorv1.EncapsulationIPIP, operatorv1.EncapsulationIPIPCrossSubnet:
+						// IPIP currently requires BGP to be running in order to program routes.
+						if instance.Spec.CalicoNetwork.BGP == nil || *instance.Spec.CalicoNetwork.BGP == operatorv1.BGPDisabled {
+							return fmt.Errorf("IPIP encapsulation requires that BGP is enabled")
+						}
+					case operatorv1.EncapsulationVXLAN, operatorv1.EncapsulationVXLANCrossSubnet:
+					case operatorv1.EncapsulationNone:
+						// Unencapsulated currently requires BGP to be running in order to program routes.
+						if instance.Spec.CalicoNetwork.BGP == nil || *instance.Spec.CalicoNetwork.BGP == operatorv1.BGPDisabled {
+							return fmt.Errorf("Unencapsulated IP pools require that BGP is enabled")
+						}
+					default:
+						return fmt.Errorf("%s is invalid for ipPool.encapsulation, should be one of %s",
+							v4pool.Encapsulation, strings.Join(operatorv1.EncapsulationTypesString, ","))
 					}
-				case operatorv1.EncapsulationVXLAN, operatorv1.EncapsulationVXLANCrossSubnet:
-				case operatorv1.EncapsulationNone:
-					// Unencapsulated currently requires BGP to be running in order to program routes.
-					if instance.Spec.CalicoNetwork.BGP == nil || *instance.Spec.CalicoNetwork.BGP == operatorv1.BGPDisabled {
-						return fmt.Errorf("Unencapsulated IP pools require that BGP is enabled")
+				} else {
+					// Verify the specified encapsulation type is None for non Calico IPAM.
+					if v4pool.Encapsulation != operatorv1.EncapsulationNone {
+						return fmt.Errorf("%s is invalid for ipPool.encapsulation with %s CNI and %s IPAM",
+							v4pool.Encapsulation,
+							instance.Spec.CNI.Type,
+							instance.Spec.CNI.IPAM.Type)
+					}
+					if instance.Spec.CalicoNetwork.BGP != nil && *instance.Spec.CalicoNetwork.BGP != operatorv1.BGPDisabled {
```

Comment:
This is the validation that I think is not correct - now that I think about it, I am certain that at least one major user is running in this mode (Calico CNI, host-local IPAM, and BGP to distribute routes)

Reply:
What about Encapsulation? If we're supporting BGP with HostLocal then shouldn't we support any encapsulation?

Reply:
IPIP should work, VXLAN won't without a bit more coding.